### PR TITLE
Background reports

### PR DESCRIPTION
**Pull request by shriharishastry**
Request has initial setup of scripts for setting reports on background mode.

### CODE REVIEW

File: frappe/desk/query_report.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 import frappe
 import os, json
+import csv
+from datetime import datetime
```
Comment (shridarpatil) [inline frappe/desk/query_report.py]: No need to import datetime

use frappe's datetime functions

check frappe.utils.data.py

File: frappe/desk/query_report.py [hunk 2]
```diff
@@ -29,6 +35,47 @@ def get_report_doc(report_name):
 
 	return doc
 
+
+def generate_report_result(report, filters, user):
```
Comment (shridarpatil) [inline frappe/desk/query_report.py]: if not user get the current session user

File: frappe/desk/query_report.py [hunk 3]
```diff
@@ -61,6 +108,57 @@ def get_script(report_name):
 		"html_format": html_format
 	}
 
+
+def create_csv_file(columns, data):
+	print columns
+	column_list = [str(d) for d in columns]  # create the list of column labels
+	csv_filename = '{0}.csv'.format(datetime.now().strftime("%Y%m%d-%H%M%S"))
```
Comment (shridarpatil) [inline frappe/desk/query_report.py]: use frappes datetime

File: frappe/desk/query_report.py (hunk 1 above)
Comment (shriharishastry) [reply frappe/desk/query_report.py]: Will make necessary changes.

File: frappe/desk/query_report.py [hunk 4]
```diff
@@ -29,6 +34,47 @@ def get_report_doc(report_name):
 
 	return doc
 
+
+def generate_report_result(report, filters):
```
Comment (shridarpatil) [inline frappe/desk/query_report.py]: add docstring

File: frappe/desk/query_report.py [hunk 5]
```diff
@@ -61,6 +107,56 @@ def get_script(report_name):
 		"html_format": html_format
 	}
 
+
+def create_csv_file(columns, data, dt, dn):
```
Comment (shridarpatil) [inline frappe/desk/query_report.py]: add docstring to every function

File: frappe/__init__.py [hunk 6]
```diff
@@ -17,7 +17,7 @@
 from .exceptions import *
 from .utils.jinja import (get_jenv, get_template, render_template, get_email_from_template, get_jloader)
 
-__version__ = '10.1.36'
+__version__ = '10.1.32'
```
Comment (shridarpatil) [inline frappe/__init__.py]: why did you change the frappe version

File: frappe/core/doctype/background_report_result/background_report_result.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+class BackgroundReportResult(Document):
+	def after_insert(self):
+		enqueue(
+			run_background,
+			instance=self, timeout=6000
+		)
+
+
+@frappe.whitelist()
```
Comment (shridarpatil) [inline frappe/core/doctype/background_report_result/background_report_result.py]: why is it whitelisted?

File: frappe/core/doctype/background_report_result/background_report_result.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+def run_background(instance):
+	report = frappe.get_doc("Report", instance.ref_report_doctype)
+	result = generate_report_result(report, filters=json.loads(instance.filters), user=instance.owner)
+	create_csv_file(result['columns'], result['result'], 'Background Report Result', instance.name)
+	instance.status = "Completed"
+	instance.report_end_time = frappe.utils.now()
+	instance.save()
+
+
+@frappe.whitelist()
```
Comment (shridarpatil) [inline frappe/core/doctype/background_report_result/background_report_result.py]: whitelist??

File: frappe/core/doctype/background_report_result/test_background_report_result.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable */
+// rename this file from _test_[name] to test_[name] to activate
+// and remove above this line
+
+QUnit.test("test: Background Report Result", function (assert) {
```
Comment (shridarpatil) [inline frappe/core/doctype/background_report_result/test_background_report_result.js]: add one test case

**Review by shridarpatil — CHANGES_REQUESTED** [review]

File: frappe/core/doctype/background_report_result/background_report_result.py [hunk 10]
```diff
@@ -9,21 +9,36 @@
 import json
 
 import frappe
+<<<<<<< HEAD
```
Comment (shridarpatil) [inline frappe/core/doctype/background_report_result/background_report_result.py]: fix merge conflicts

File: frappe/core/doctype/background_report_result/background_report_result.py [hunk 11]
```diff
@@ -9,21 +9,36 @@
...
 from frappe.utils.background_jobs import enqueue
 from frappe.desk.query_report import generate_report_result, get_columns_dict
 from frappe.utils.file_manager import save_file
 
 
 class BackgroundReportResult(Document):
+<<<<<<< HEAD
+
+	def before_insert(self):
+		self.status = "Queued"
```
Comment (shridarpatil) [inline frappe/core/doctype/background_report_result/background_report_result.py]: fix merge conflicts

File: frappe/core/doctype/background_report_result/test_background_report_result.py [hunk 12]
```diff
@@ -5,6 +5,36 @@
...
+		}
+		self.background_report_doc = frappe.get_doc({
+			"doctype": "Background Report Result",
+			"report_name": self.report.name,
+			"filters": json.dumps(self.filters),
+			"ref_report_doctype": self.report.name,
+			"report_type": self.report.report_type,
+			"query": self.report.query,
+			"module": self.report.module
+		}) \
```
Comment (shridarpatil) [inline frappe/core/doctype/background_report_result/test_background_report_result.py]: why . insert in next line?
add it to same line, it adds unnecessary new line

**Review by shridarpatil — CHANGES_REQUESTED** [review]

File: frappe/core/doctype/background_report_result/test_background_report_result.py (hunk 12 above)
Comment (shriharishastry) [reply frappe/core/doctype/background_report_result/test_background_report_result.py]: Requested changes are done.

File: frappe/core/doctype/background_report_result/background_report_result.py (hunk 8 above)
Comment (shriharishastry) [reply frappe/core/doctype/background_report_result/background_report_result.py]: Requested changes are done.

File: frappe/core/doctype/background_report_result/background_report_result.py (hunk 7 above)
Comment (shriharishastry) [reply frappe/core/doctype/background_report_result/background_report_result.py]: Requested changes are done.
